### ISSUE TITLE
media links lose the protocol part on v4.6.5

### ISSUE THREAD

**Issue by eiennohi** [tinymce/tinymce#3889]
**When embedding a video through the media button on the toolbar, the URL loses the protocol part**

my tinymce config:
```
tinymce.init({
      selector:'my-tinymce > textarea',
      autoresize_max_height: '400',
      content_css : "/stylesheets/main.css",
      format: 'raw',
      menubar: false,
      toolbar: "styleselect | bold italic underline textcolor | alignleft aligncenter alignright alignjustify | bullist numlist | link image media | removeformat | code",
      statusbar: false,
      relative_urls: false,
      remove_script_host: false,
      urlconverter_callback: function(url, node, on_save, name) {
        console.warn('url: ', url);
        return url;
      },
      plugins: [
	         "advlist autolink link image lists preview hr anchor pagebreak",
	         "media nonbreaking code",
	         "table directionality autoresize"
	    ],
      setup: (ed) => {

        (...)

      }
    });
```
As you can see from my config, I'm logging the url exactly when urlconverter_callback is called. This is the output:
![image](https://user-images.githubusercontent.com/6691897/29719693-919f1122-898d-11e7-926a-8e7c7a310444.png)

The first three are from the moment I pasted the URL onto the Source input. The other ones appear right after hitting the OK button.

**Steps to reproduce:**

1. open the media modal
2. paste the URL onto source
3. hit OK

Happened with TinyMCE v4.5.6 and 4.2.6

**Browsers:**
- Chrome 60.0.3112.101
- Firefox 54.0.1

**Comment by spocke** [comment]
This is by design the urls are protocol relative so that they work on http and https depending on what your site has. I guess we could force them to always be https.

**Comment by Schenn** [comment]
Please don't force any protocols.  We may not always be linking to video content which is relative to our site's protocols. (e.g. https logged in page to an http hosted video).  Can't you just see if there's a protocol present in the url, if so leave it. If not, add the generic http? That way, we as developers can have expected outcomes when using this.

**Comment by lnewson** [comment]
Hi,

This was actually implemented in TinyMCE 5.2.0, which was released last month. The media plugin will now default to using https unless otherwise specified in the URL provided. If you encounter any other issues, please let us know.

Thanks!
